### PR TITLE
HAI-1329 Update React

### PR DESCRIPTION
**Pull request by markohaarni**
# Description

Updated react and react-dom to version 18.2.0. Also updated many of the other dependencies as well.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [x] Other

# Instructions for testing

Please describe tests how this change or new feature can be tested.

# Checklist:

- [ ] I have written new tests (if applicable)
- [ ] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

# Other relevant info

Please describe here if there is e.g. some requirements for this change or
other info that the tester/user needs to know.

### CODE REVIEW

File: src/domain/forms/components/Contact.tsx [hunk 1]
```diff
@@ -17,7 +16,7 @@ interface Props<T> {
   children: React.ReactNode;
 }
 
-const Contact = <T extends unknown>({
+const Contact = <T,>({
```
Comment (pitkni) [inline src/domain/forms/components/Contact.tsx:19]: Is the comma a typo or is it really needed?

Comment (markohaarni) [reply src/domain/forms/components/Contact.tsx:19]: Yes, it's needed.

**Review by pitkni — CHANGES_REQUESTED** [review]
Dockerfile-local needs to update node version.

**Comment by markohaarni** [comment]
> Dockerfile-local needs to update node version.

Node version updated

**Review by pitkni — APPROVED** [review]

**Pull request merged**